### PR TITLE
Fix #26846: Edit Ema copy url and api link buttons

### LINKED ISSUE #26846: Edit EMA: Copy URL and API Link Buttons
### Parent
#26644 

### User Story 
As a dotCMS user, I want to be able to quickly copy the URL of a page and have a quick link to the Page API response, so that I can quickly access and share the URLs.

```[tasklist]
### Acceptance Criteria
- [x] The user should be able to copy the URL of a page with a single click.
- [x] A toast should be displayed when the URL has been successfully copied.
- [x] The user should be able to get a quick link to the Page API response with a single click.
```

https://github.com/dotCMS/core/assets/751424/adf314dc-a32d-4957-9e95-ec552aa3ef74

### Proposed Objective
Core Features

### Proposed Priority
Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.:
N/A

### Assumptions & Initiation Needs
N/A

### Quality Assurance Notes & Workarounds
N/A

### PR DESCRIPTION
**Pull request by zJaaal**
### Summary
🚚🎨🛠️

This pull request updates the style and functionality of some components related to the dot-ema portlet. It adds a new persona selector component that uses a listbox and an avatar directive. It also enhances the language selector component to use a listbox and a native button. It moves the avatar directive to a new location and updates the imports in several files. It changes the border color of the toolbar component and removes some unused styles.

> _We're moving `DotAvatarDirective` to a new place_
> _To make our code more tidy and easy to trace_
> _So heave away, me hearties, on the count of three_
> _And we'll update the toolbar and the language selector, aye!_

### Walkthrough
*  Moved the import of `DotAvatarDirective` from `@directives/dot-avatar/dot-avatar.directive` to `@dotcms/ui` in several files to use the new location of the directive ([link](https://github.com/dotCMS/core/pull/26905/files?diff=unified&w=0#diff-413a52aa4eac9e5381f557c3c283c4d56952ec34573504c0d80626a8d849884dL11-R14), [link](https://github.com/dotCMS/core/pull/26905/files?diff=unified&w=0#diff-c1500b5c5554ac0cff95a496a7bdc7fc644cd5f6fb91122f9d076533bde049e4L9-R9), [link](https://github.com/dotCMS/core/pull/26905/files?diff=unified&w=0#diff-2e72fc43eb122febc6b45ee93c35f7d544a62acac99226648f70419db613c016L10-R11), [link](https://github.com/dotCMS/core/pull/26905/files?diff=unified&w=0#diff-c4dbb5a3335a72b4552f79fc0f913d4b07037c3b7cd5021fed404647931bed98L11-R11), [link](https://github.com/dotCMS/core/pull/26905/files?diff=unified&w=0#diff-4acb8f72b293413c8643f47fc40fe763c4437b839ad287d06792b247dda13eacL10-R12), [link](https://github.com/dotCMS/core/pull/26905/files?diff=unified&w=0#diff-8b48ca47b42209ffde90fb3a9ffa91ad8c20068ce7b28c464db93e9210cfe062L10-R10), [link](https://github.com/dotCMS/core/pull/26905/files?diff=unified&w=0#diff-a5899873377b6023863f10ea7c26baa39867901fc0361e64cefd81c1ea211e5fL10-R11), [link](https://github.com/dotCMS/core/pull/26905/files?diff=unified&w=0#diff-7207aae5cdcc2225ebfa4b269da518b301fec3cc213661e146ead50bd0801810L9-R9)).

### Screenshots

https://github.com/dotCMS/core/assets/63567962/577e08e7-63dd-401d-86e3-56d964604f5e

### CODE REVIEW

File: core-web/libs/portlets/dot-ema/src/lib/components/edit-ema-language-selector/edit-ema-language-selector.component.ts [hunk 1]
```diff
@@ -21,7 +29,8 @@ interface DotLanguageWithLabel extends DotLanguage {
     templateUrl: './edit-ema-language-selector.component.html',
     styleUrls: ['./edit-ema-language-selector.component.scss']
```
Comment (oidacra) [inline core-web/libs/portlets/dot-ema/src/lib/components/edit-ema-language-selector/edit-ema-language-selector.component.ts:30]: You dont need here OnPush?

File: core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.html [hunk 2]
```diff
@@ -1,19 +1,38 @@
...
-        <h4 title>{{ vm.pageTitle }}</h4>
         <ng-container left>
-            <div class="dummy-button"></div>
-            <div class="dummy-button"></div>
-            <div class="dummy-button"></div>
-            <div class="dummy-button"></div>
+            <p-button label="Preview" icon="pi pi-desktop" styleClass="p-button-text p-button-sm" />
+            <p-button label="Bookmark" icon="pi pi-star" styleClass="p-button-text p-button-sm" />
+            <p-button
+                [cdkCopyToClipboard]="vm.iframeUrl"
```
Comment (fmontes) [inline core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.html:7]: This directive have a `copied` event, check here: https://material.angular.io/cdk/clipboard/api#directives

File: core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.html [hunk 3]
```diff
@@ -1,19 +1,38 @@
...
+                [label]="'editpage.header.copy' | dm"
+                (onClick)="triggerCopyToast()"
+                icon="pi pi-copy"
+                styleClass="p-button-text p-button-sm"
+                data-testId="ema-copy-url" />
+            <a
+                class="p-button-sm p-button-text"
+                [href]="vm.apiURL"
+                [title]="vm.apiURL"
+                pButton
```
Comment (fmontes) [inline core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.html:17]: If this works... why do we need to add the span manually?

File: core-web/libs/portlets/dot-ema/src/lib/feature/store/dot-ema.store.ts [hunk 4]
```diff
@@ -61,15 +66,29 @@ export class EditEmaStore extends ComponentStore<EditEmaState> {
...
+                  dialogHeader: state.dialogHeader,
+                  dialogIframeLoading: state.dialogIframeLoading,
+                  editor: state.editor,
+                  selectedPersona: state.editor.viewAs.persona ?? DEFAULT_PERSONA,
+                  apiURL: `${window.location.origin}/api/v1/page/render/${this.createPageURL({
+                      url: state.url,
+                      language_id: state.editor.viewAs.language.id.toString(),
+                      persona_id:
+                          state.editor.viewAs.persona?.identifier ?? DEFAULT_PERSONA.identifier
+                  })}`
```
Comment (fmontes) [inline core-web/libs/portlets/dot-ema/src/lib/feature/store/dot-ema.store.ts:89]: WOW... let's clean this state..

`pageTitle` and `selectedPersona`, `apiURL` comes inside the `editor` we can get this from the state with a selector and some mapping... let's try to keep this state clean and slim.

Comment (fmontes) [reply core-web/libs/portlets/dot-ema/src/lib/feature/store/dot-ema.store.ts:89]: Basically anything that is already in the state in properties I don't think we should duplicated to make it surface, is better just to use selectors.

File: core-web/libs/portlets/dot-ema/src/lib/shared/models.ts [hunk 5]
```diff
@@ -3,6 +3,7 @@ export interface AddContentletPayload {
     container?: Container;
     pageContainers: Container[];
     pageID: string;
+    personaTag?: string;
```
Comment (fmontes) [inline core-web/libs/portlets/dot-ema/src/lib/shared/models.ts:6]: Are you sure this is optional?

**Review by fmontes — CHANGES_REQUESTED** [review]

**Review by fmontes — APPROVED** [review]

**Review by manuelrojas — APPROVED** [review]

**Review by rjvelazco — APPROVED** [review]

**Pull request merged**